### PR TITLE
MM-20934: Fixing int overflow in 32 bits on MaxImageSize check

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fixing int overflow in 32 bits on MaxImageSize check. The width/heigh was
working with big-enough numbers for handling huge images in 64 bits, but in 32
bits, there were using 32 bits to represent the width and the height (what is
enough) but not when you multiple them. This was generating and overflow that
can lead to allow you to upload images bigger than the allowed ones.

I tested the solution in ARM and 386 architectures (only the overflow situation, not whole upload image process).

#### Ticket Link
[MM-20934](https://mattermost.atlassian.net/browse/MM-20934)

### CODE REVIEW

**Comment by jespino** [comment]
@lindy65 about testing it probably @jupenur can help, because he was who reported the bug in first place, so probably he have a system where he can test the fix somehow.

File: app/brand.go [hunk 1]
```diff
@@ -38,7 +38,10 @@ func (a *App) SaveBrandImage(imageData *multipart.FileHeader) *model.AppError {
 		return model.NewAppError("SaveBrandImage", "brand.save_brand_image.decode_config.app_error", nil, err.Error(), http.StatusBadRequest)
 	}
 
-	if config.Width*config.Height > model.MaxImageSize {
+	// This casting is done to prevent overflow on 32 bit systems (not needed
+	// in 64 bits systems because images can have more than 32 bits height or
```
Comment (jupenur) [inline app/brand.go:42]:
```suggestion
	// in 64 bits systems because images can't have more than 32 bits height or
```

File: app/file.go [hunk 2]
```diff
@@ -728,7 +728,10 @@ func (t *UploadFileTask) preprocessImage() *model.AppError {
 	t.fileinfo.Height = config.Height
 
 	// Check dimensions before loading the whole thing into memory later on.
-	if t.fileinfo.Width*t.fileinfo.Height > MaxImageSize {
+	// This casting is done to prevent overflow on 32 bit systems (not needed
+	// in 64 bits systems because images can have more than 32 bits height or
```
Comment (jupenur) [inline app/file.go:732]:
```suggestion
	// in 64 bits systems because images can't have more than 32 bits height or
```

File: app/file.go [hunk 3]
```diff
@@ -911,7 +914,10 @@ func (a *App) DoUploadFileExpectModification(now time.Time, rawTeamId string, ra
 
 	if info.IsImage() {
 		// Check dimensions before loading the whole thing into memory later on
-		if info.Width*info.Height > MaxImageSize {
+		// This casting is done to prevent overflow on 32 bit systems (not needed
+		// in 64 bits systems because images can have more than 32 bits height or
```
Comment (jupenur) [inline app/file.go:918]:
```suggestion
		// in 64 bits systems because images can't have more than 32 bits height or
```

File: app/team.go [hunk 4]
```diff
@@ -1510,7 +1510,11 @@ func (a *App) SetTeamIconFromMultiPartFile(teamId string, file multipart.File) *
 	if err != nil {
 		return model.NewAppError("SetTeamIcon", "api.team.set_team_icon.decode_config.app_error", nil, err.Error(), http.StatusBadRequest)
 	}
-	if config.Width*config.Height > model.MaxImageSize {
+
+	// This casting is done to prevent overflow on 32 bit systems (not needed
+	// in 64 bits systems because images can have more than 32 bits height or
```
Comment (jupenur) [inline app/team.go:1515]:
```suggestion
	// in 64 bits systems because images can't have more than 32 bits height or
```

File: app/user.go [hunk 5]
```diff
@@ -833,7 +833,10 @@ func (a *App) SetProfileImageFromMultiPartFile(userId string, file multipart.Fil
 	if err != nil {
 		return model.NewAppError("SetProfileImage", "api.user.upload_profile_user.decode_config.app_error", nil, err.Error(), http.StatusBadRequest)
 	}
-	if config.Width*config.Height > model.MaxImageSize {
+	// This casting is done to prevent overflow on 32 bit systems (not needed
+	// in 64 bits systems because images can have more than 32 bits height or
```
Comment (jupenur) [inline app/user.go:837]:
```suggestion
	// in 64 bits systems because images can't have more than 32 bits height or
```

File: app/brand.go (hunk 1 above)
Comment (jespino) [reply app/brand.go]: Thanks :facepalm:

**Comment by jespino** [comment]
@jupenur I need your approval here to be able to merge this?

**Pull request merged**